pkg/network/usm/sharedlibraries/watcher_test.go

```diff
@@ -94,42 +75,17 @@ func (s *SharedLibrarySuite) TestSharedLibraryDetection() {
 	launchProcessMonitor(t)
 
 	// create files
-	clientBin := buildSOWatcherClientBin(t)
-	command1 := exec.Command(clientBin, fooPath1)
-	require.NoError(t, command1.Start())
-	registerProcessTerminationUponCleanup(t, command1)
+	command1 := fileopener.OpenFromAnotherProcess(t, fooPath1)
 
 	require.Eventuallyf(t, func() bool {
-		// Checking path1 still exists, and path2 not.
-		if checkPathIDDoesNotExist(watcher, fooPathID1) || checkPIDNotAssociatedWithPathID(watcher, fooPathID1, uint32(command1.Process.Pid)) {
-			return false
-		}
-
-		// Checking PID1 is not associated to the path 2, and PID2 is associated only with the path2
-		return fooPath1 == pathDetected
+		return registerRecorder.CallsForPathID(fooPathID1) == 1
 	}, time.Second*10, time.Second, "")
 
 	require.NoError(t, command1.Process.Kill())
 
 	require.Eventuallyf(t, func() bool {
-		// Checking path1 still exists, and path2 not.
-		return checkPathIDDoesNotExist(watcher, fooPathID1) && checkPIDNotAssociatedWithPathID(watcher, fooPathID1, uint32(command1.Process.Pid))
+		return unregisterRecorder.CallsForPathID(fooPathID1) == 1
 	}, time.Second*10, time.Second, "")
-
-	tel := telemetry.ReportPayloadTelemetry("1")
-	telEqual := func(t *testing.T, expected int64, m string) {
-		require.Equal(t, expected, tel[m], m)
-	}
-	require.GreaterOrEqual(t, tel["usm.so_watcher.hits"], tel["usm.so_watcher.matches"], "usm.so_watcher.hits")
-	telEqual(t, 0, "usm.so_watcher.already_registered")
-	telEqual(t, 0, "usm.so_watcher.blocked")
-	telEqual(t, 1, "usm.so_watcher.matches")
-	telEqual(t, 1, "usm.so_watcher.registered")
-	telEqual(t, 0, "usm.so_watcher.unregister_errors")
-	telEqual(t, 1, "usm.so_watcher.unregister_no_callback")
-	telEqual(t, 0, "usm.so_watcher.unregister_failed_cb")
-	telEqual(t, 0, "usm.so_watcher.unregister_pathid_not_found")
-	telEqual(t, 1, "usm.so_watcher.unregistered")
```

Comment:
why don't you check the telemetry?

Reply:
the comment is applied to the entire file

Reply:
Because telemetry no longer belongs to the `sharedlibraries.Watcher`